plusplusbot/analytics/leaderboard.py

```diff
@@ -0,0 +1,111 @@
+# Thanks to https://github.com/BeheadedKamikaze for his help with the algorithm
+import logging
+import math
+from collections import defaultdict
+
+import pendulum
+
+
+class LeaderBoard:
+
+    NOT_FOUND = -1
+
+    def __init__(self, history):
+        self.history = history
+        self.len = len(history)
+        self.logger = logging.getLogger('PlusPlusBot.analytics.LeaderBoard')
+
+    # First step, find something in the history that matches the range.
+    def find_rows_in_range(self, start_time, end_time):
```

Comment:
Are you able to do a bit of a docstring around the logic of the algorithm here? I kinda get it, but the fine tuning would be nice to explain a bit more